ibis/expr/tests/test_newrels.py

```diff
@@ -561,6 +560,26 @@ def test_join():
         )
 
 
+def test_join_integrity_checks():
+    t1 = ibis.table(name="t1", schema={"a": "int64", "b": "string"})
+
+    # correct example
+    r1 = ops.JoinTable(t1, 10).to_expr()
+    r2 = ops.JoinTable(t1, 20).to_expr()
+    values = {"a": r1.a, "b": r1.b, "c": r2.a, "d": r2.b}
+    chain = ops.JoinChain(
+        r1, [ops.JoinLink("inner", r2, [r1.a == r2.a])], values=values
+    )
+    assert isinstance(chain, JoinChain)
+
+    # not unique tables
+    r1 = ops.JoinTable(t1, 10).to_expr()
+    r2 = ops.JoinTable(t1, 10).to_expr()
+    values = {"a": r1.a, "b": r1.b, "c": r2.a, "d": r2.b}
+    with pytest.raises(IntegrityError):
+        ops.JoinChain(r1, [ops.JoinLink("inner", r2, [r1.a == r2.a])], values=values)
```

Comment:
Can you construct only the exact thing that raising here? It's hard to tell what's raising just by reading the code.

Reply:
Updating.

Reply:
Reduced the example as much as I can.